### PR TITLE
Bug fixes for `sky_snapshot --compile-all` mode

### PR DESCRIPTION
**Pull request by johnmccutchan**
@chinmaygarde

### CODE REVIEW

File: snapshotter/main.cc [hunk 1]
```diff
@@ -121,6 +122,49 @@ Dart_Handle HandleLibraryTag(Dart_LibraryTag tag,
...
+    DART_CHECK_VALID(library);
+    Dart_NativeEntryResolver old_resolver = NULL;
+    Dart_GetNativeResolver(library, &old_resolver);
+    if (old_resolver == NULL) {
+      Dart_Handle result =
+          Dart_SetNativeResolver(library, &StubNativeLookup, &StubNativeSymbol);
+      DART_CHECK_VALID(result);
+    }
+  }
+}
```
Comment (abarth) [inline snapshotter/main.cc]: This stuff should probably be in tonic.  It's not really related to Flutter.

File: snapshotter/main.cc [hunk 2]
```diff
@@ -121,6 +122,49 @@ Dart_Handle HandleLibraryTag(Dart_LibraryTag tag,
...
+
+// Registers a dummy native symbol resolver on all loaded libraries that do not
+// already have a native symbol resolver. This is necessary because
+// `--compile-all` will try and resolve all native functions. The function
+// returned by the dummy native symbol resolver will never be invoked.
+static void SetupStubNativeResolvers() {
+  Dart_Handle library_ids = Dart_GetLibraryIds();
+  intptr_t library_ids_length;
+  Dart_ListLength(library_ids, &library_ids_length);
```
Comment (chinmaygarde) [inline snapshotter/main.cc]: Missing error checking.

File: snapshotter/main.cc (hunk 1 above)
Comment (johnmccutchan) [inline snapshotter/main.cc]: This is specific to the snapshotter and not something that will be reused.

File: snapshotter/main.cc (hunk 2 above)
Comment (chinmaygarde) [inline snapshotter/main.cc]: I guess error checking is not performed for any of these calls. I'd rather escape out of this routine on any error. Maybe a macro or something. But if you think that is being too paranoid, I understand. I don't know how these `dart_api.h` calls can fail.

Comment (johnmccutchan) [inline snapshotter/main.cc]: I'll wrap these calls with DART_CHECK_VALID.

Comment (johnmccutchan) [inline snapshotter/main.cc]: Fixed.

**Pull request merged**